ggml-cuda.cu

```diff
@@ -8661,11 +8660,26 @@ static void ggml_cuda_mul_mat(const ggml_tensor * src0, const ggml_tensor * src1
         }
     }
 
-#ifdef CUDA_USE_TENSOR_CORES
-    const bool use_tensor_cores = true;
+#if defined(GGML_USE_HIPBLAS) && defined(__HIP_PLATFORM_AMD__)
+    const bool fp16_performance_good = true;
+
+#ifdef RDNA3
+    const bool use_mul_mat_q = false;
 #else
-    const bool use_tensor_cores = false;
-#endif
+    const bool use_mul_mat_q = true;
+#endif // RDNA3
+
+#else
+
+    const bool fp16_performance_good = min_compute_capability >= CC_VOLTA;
```

Comment:
A minor nit. FP16 performance on Pascal is weird. They do not have tensorcores, so it's not going to be great, but there may still be some benefit in using it on  sm_60 (P100). sm_61 (P10, RTX* cards) indeed has horrible fp16 performance and things are way faster in fp32, however, on sm_60 you may get up to 2x peak performance improvements in fp16 vs fp32.

It may be interesting to benchmark performance on P100 with fp16 enabled.

Reply:
That's super strange. SM_61 has good INI8 perf. maybe this is the cause?

Reply:
I think for integers, performance will be comparable. It's specifically fp16 that behaves differently.

Reply:
Do you have access to a P100? If yes I can do a quick PR to try and fix this and you can check whether it actually works.

Reply:
Not directly. Google cloud should still have some available. https://cloud.google.com/compute/docs/gpus

Reply:
I think kaggle have a free P100 as well for 30hour each week